src/api.rs

```diff
@@ -76,3 +76,219 @@ impl Api {
         Ok(Json(plan))
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
@coderabbitai Following the pattern for existing api unit test in `api.rs`, please generate additional unit tests for:
- posts to `/plan` (aka `plan_post()`)
- gets to `/plan/:dataset_id` (aka `plan_get()`)
- puts to `/data_product/:dataset_id` (aka `state_put()`)